kernel/src/hil/time.rs

```diff
@@ -175,3 +177,50 @@ pub trait TimerClient {
     /// Callback signaled when the timer's clock reaches the specified interval.
     fn fired(&self);
 }
+
+pub trait TimerWrappingOps {
+    fn wrapping_add(x: u32, y: u32) -> u32;
+
+    fn wrapping_sub(x: u32, y: u32) -> u32;
+}
+
+#[derive(Debug)]
+pub struct Timer32Bit;
+impl TimerWrappingOps for Timer32Bit {
+    fn wrapping_add(x: u32, y: u32) -> u32 {
+        x.wrapping_add(y)
+    }
+
+    fn wrapping_sub(x: u32, y: u32) -> u32 {
+        x.wrapping_sub(y)
+    }
+}
+
+#[derive(Debug)]
+pub struct Timer24Bit;
+impl TimerWrappingOps for Timer24Bit {
```

Comment:
I believe the correctness of these implementations rely on `x` and `y` actually being 24 bits, and could re-introduce wrapping issues if they're actually 32 bits (as the type signature unavoidable indicates). I don't think this is a problem in practice, but wanted to flag it